utils/sample_qc.py

```diff
@@ -364,91 +364,109 @@ def infer_sex(
-    :param Tuple[float, float] xx_x_ploidy_cutoffs: Boundaries around the chom X ploidy for females
-    :param Tuple[float, float] xy_x_ploidy_cutoffs: Boundaries around the chom X ploidy for males
-    :param float xx_y_ploidy_cutoff: Max y ploidy for females
-    :param Tuple[float, float] xy_y_ploidy_cutoffs: Boundaries around the chom Y ploidy for males
-    :param float yy_y_ploidy_cutoff: Min chrom Y ploidy for YY
-    :param float xxx_x_ploidy_cutoff: Min chrom X ploidy for XXX
-    :param float f_stat_female_cutoff: Min f-stat for females
-    :param flat f_stat_male_cutoff: Max f-stat for males
-    :return: Struct expression with sex anotations
+    :param Table ht: Table with samples, their imputed sex, and their sex chromosome ploidies
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, sex: str) -> dict:
+        """
+        Gets chromosome X and Y ploidy cutoffs for XY and XX samples
+
+        :param Table ht: Table with samples, their imputed sex, and their sex chromosome ploidies
+        :param str sex: 'XY' or 'XX'
+        :return: Dictionary of ploidy cutoffs; key: sex chrom, values: [lower cut, upper cut, outlier (aneuploidy) cut]
+        :rtype: dict
+        """
+        if sex == "XY":
+            ht = ht.filter((hl.is_defined(ht.is_female)) & (~ht.is_female))
+        else:
+            ht = ht.filter(ht.is_female)
+
+        # Get mean/stdev for chrX/Y ploidies
+        X_stats = ht.aggregate(hl.agg.stats(ht.chrX_ploidy))
+        Y_stats = ht.aggregate(hl.agg.stats(ht.chrY_ploidy))
+
+        # Set cutoffs -- keep cutoffs permissive to account for loss of sex chromosomes
+        # NOTE: keep lower bound of Y particularly permissive (0.05) to save males with somatic loss of Y 
+        cutoffs = {}
+        cutoffs["X"] = [X_stats.mean - (5 * X_stats.stdev), X_stats.mean + (5 * X_stats.stdev), X_stats.mean + (6 * X_stats.stdev)]
+        cutoffs["Y"] = [min(0.05, Y_stats.mean - (6 * Y_stats.stdev)), Y_stats.mean + (5 * Y_stats.stdev), Y_stats.mean + (6 * Y_stats.stdev)]
+        return cutoffs
+   
+ 
+    xx_cutoffs = get_ploidy_cutoffs(ht, "XX")
+    xy_cutoffs = get_ploidy_cutoffs(ht, "XY")
+
+    xx_x_ploidy_cutoffs = xx_cutoffs["X"][:2]
+    xx_y_ploidy_cutoff = xx_cutoffs["Y"][2]
+    xxx_x_ploidy_cutoff = xx_cutoffs["X"][2]
+    logger.info('XX cutoffs:')
+    logger.info(f'X {xx_x_ploidy_cutoffs}')
+    logger.info(f'Y {xx_y_ploidy_cutoff}')
+    logger.info(f'XXX {xxx_x_ploidy_cutoff}')
+
+    xy_x_ploidy_cutoff = xy_cutoffs["X"][1]
+    xy_y_ploidy_cutoffs = xy_cutoffs["Y"][:2]
+    yy_y_ploidy_cutoff = xy_cutoffs["Y"][2]
+    logger.info('XY cutoffs:')
+    logger.info(f'X {xy_x_ploidy_cutoff}')
+    logger.info(f'Y {xy_y_ploidy_cutoffs}')
+    logger.info(f'YY {yy_y_ploidy_cutoff}')
 
     sex_expr = hl.struct(
         sex_karyotype=(
             hl.case()
                 .when(
-                (chr_x_ploidy > xx_x_ploidy_cutoffs[0]) &
```

Comment:
I wonder if this could be restructured as something like
````
hl.case().when(..., 'X').when(..., 'XX').when(..., 'XXX') +
hl.case().when(..., 'Y').when(..., 'YY')
```
instead of having each and every X, Y possibility spelled out.